remove the niri-session, as that's a systemd-only binary. replace with niri --session which automatically adds envs.
--
--- a/resources/niri.desktop
+++ b/resources/niri.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=Niri
 Comment=A scrollable-tiling Wayland compositor
-Exec=niri-session
+Exec=/usr/bin/niri --session
 Type=Application
 DesktopNames=niri
